src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -470,5 +493,58 @@ public bool TryGetDateTimeOffset(out DateTimeOffset value)
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
             return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
         }
+
+        /// <summary>
+        /// Reads the next JSON token value from the source and parses it to a <see cref="Guid"/>.
+        /// Returns <see langword="true"/> if the entire UTF-8 encoded token value can be successfully
+        /// parsed to a <see cref="Guid"/> value. Only supports <see cref="Guid"/> values with hyphens
+        /// and without any surrounding decorations.
+        /// Returns <see langword="false"/> otherwise.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if trying to get the value of a JSON token that is not a <see cref="JsonTokenType.String"/>.
+        /// <seealso cref="TokenType" />
+        /// </exception>
+        public bool TryGetGuid(out Guid value)
+        {
+            if (TokenType != JsonTokenType.String)
+            {
+                throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
+            }
+
+            long sequenceLength;
+            if (HasValueSequence && ((sequenceLength = ValueSequence.Length)  <= JsonConstants.MaximumEscapedGuidLength))
+            {
+                Span<byte> span = stackalloc byte[(int)sequenceLength];
+                ValueSequence.CopyTo(span);
+
+                if (_stringHasEscaping)
+                {
+                    return JsonReaderHelper.TryGetEscapedGuid(span, out value);
+                }
+
+                Debug.Assert(span.IndexOf(JsonConstants.BackSlash) == -1);
+
+                value = default;
+                return (span.Length == JsonConstants.MaximumFormatGuidLength) && Utf8Parser.TryParse(span, out value, out _, 'D');
```

Comment:
nit: A lot of this code is repeated between the branches.

How about something like this, in an attempt to reduce the duplication:
```C#
            ReadOnlySpan<byte> span = ValueSpan;
            Span<byte> stackSpan;

            if (HasValueSequence)
            {
                long sequenceLength = ValueSequence.Length;
                if (sequenceLength > JsonConstants.MaximumEscapedGuidLength)
                {
                    value = default;
                    return false;
                }

                // Cannot create a span directly since it gets passed to instance methods on a ref struct.
                unsafe
                {
                    byte* ptr = stackalloc byte[(int)sequenceLength];
                    stackSpan = new Span<byte>(ptr, (int)sequenceLength);
                }
                ValueSequence.CopyTo(stackSpan);
                span = stackSpan;
            }

            if (span.Length > JsonConstants.MaximumEscapedGuidLength)
            {
                value = default;
                return false;
            }

            if (_stringHasEscaping)
            {
                return JsonReaderHelper.TryGetEscapedGuid(span, out value);
            }

            Debug.Assert(span.IndexOf(JsonConstants.BackSlash) == -1);
            return Utf8Parser.TryParse(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
```

Also, note the condition reversal at the end (`Utf8Parser`) first, and then the length check. It avoids having to set value to default but maybe trying to avoid that is unnecessary.

So, it's probably better to keep it as (even thought the common case would be `Utf8Parser.TryParse` gets called):
```C#
value = default;
return (span.Length == JsonConstants.MaximumFormatGuidLength) && Utf8Parser.TryParse(span, out value, out int bytesConsumed);
```

Reply:
Nice, thank you! 
Keeping the length check, but as a standalone `if` branch so we don't always have to set `value` to `default`.

```C#
if (span.Length != JsonConstants.MaximumFormatGuidLength)
{
    value = default;
    return false;
}

return Utf8Parser.TryParse(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
```